02.calendar/cal.rb

```diff
@@ -0,0 +1,254 @@
+ # 
+ # 引数
+ # month: 数字表記の月
+ #
+ # For examlpe:
+ #  convert_month_to_english("1")
+ #   => Jan
+ def convert_month_to_english(month)
+
+  # month_nameのハッシュデータのkeyを探索する
+  $month_name[month.to_s]
+
+ end
+
+ # カレンダーの曜日をコンソールに表示する
+ #
+ # For example:
+ #   print_all_weekdate
+ #      => Sun Mon Tue Wed Thr Fri Sat
+ def print_all_weekdate
+  $week_date.each_key do |key|
+   print $week_date[key] + " "
+  end
+  puts ""
+ end
+
+ # カレンダー表示で改行される日付かどうかの判定をする
+ # 引数
+ #  first_date: カレンダー表示する月の1日にあたる曜日
+ #  date: 判定対象の日付
+ #
+ # For example:
+ #    check_newline(1,1)
+ #      => false 
+ #         # first_date == 1 のとき、表示月の1日は月曜日である。
+ #           その場合、土曜日は 7 % date == 6 のときであり、そのときに改行する。    
+ #    check_newline(3,4)
+ #      => True
+ #         # first_date == 3 のとき、表示月の1日は水曜日である。
+ #           その場合、土曜日は 7 % date == 4 のときである、そのときに改行する。
+ def check_newline(first_date, date)
+  date_mod = date % 7
+  if first_date == 1
+    if date_mod == 6
+      return true
+    end
+  elsif first_date == 2
+    if date_mod == 5
+      return true
+    end
+  elsif first_date == 3
+    if date_mod == 4
+      return true
+    end
+  elsif first_date == 4
+    if date_mod == 3
+      return true
+    end
+  elsif first_date == 5
+    if date_mod == 2
+      return true
+    end
+  elsif first_date == 6
+    if date_mod == 1
+      return true
+    end
+  elsif first_date == 0
+    if date == 1
+      return false
+    elsif date_mod == 0
+      return true
+    end
+  end
+
+  return false
+ end
+
+ # カレンダーの日付を整形して表示する
+ #
+ # 引数
+ # date: 指定した月の曜日の整数表記
+ # last_day: 表示する月の最終日(28 or 29 or 30 or 31)
+ # flg_year: 表示するカレンダーの年がオプションで指定されているかの判定フラグ
+ # flg_month: 表示するカレンダーの月がオプションで指定されているかの判定フラグ
+ #
+ #
+ def print_date(date, last_day, flg_year, flg_month)
+
+  # 開始位置の設定
+  set_first_position(date)
+  
+  # 今日の日付のフラグ
+  flg_today = false
+	
+  (1..last_day).each do |i|
+   
+   # オプションにかかわるフラグ
+   if flg_year and flg_month
+    if i == @day
+     flg_today = true
+    else
+     flg_today = false
+    end
+   end
+
+    # デフォルト出力フォーマット
+    print_str = " " + i.to_s
+    
+    # 今日日付の場合
+    if flg_today
+      print_str = ANSI_ESC + print_str + ANSI_ESC_END
+    end
+    
+    # 表示位置を調整するための半角スペースを末尾に付ける
+    print_str = print_str + " "
+    
+    # 日付が一桁の場合、表示位置を調整するため先頭に半角スペースをつける
+    if i < 10
+      print_str = " " + print_str
+    end
+    
+    # 改行が必要の場合
+    if check_newline(date,i)
+      print_str = print_str.rstrip + "\n"
+    end
+    
+    print print_str
+
+  end
+  puts ""
+ end
+
+
+ # 整形後のカレンダーを出力する
+ #
+ # 引数
+ # 
+ def print_calender
+
+  # カレンダーのタイトル表示
+  puts "        #{convert_month_to_english(@month)} #{@year}"
+
+  # 曜日表示
+  print_all_weekdate
+
+  last_day = Date.new(@year, @month, -1).day
+
+  # 日付表示
+  print_date(get_first_wday(), last_day, @flg_year, @flg_month)
+ end
+
+ # １日の表示位置までスペースで埋める
+ #
+ # 引数
+ # date: 数字表記の曜日
+ def set_first_position(date)
+   print "    " * date
+ end
+end
+
+
+# 今日の日付のオブジェクト
+day = Date.today
```

Comment:
今日の日付であれば`today`みたいな変数名の方が良さそうです。

Reply:
```suggestion
today = Date.today
```
上記のコードであれば、今日の日付のオブジェクトを`today`に格納しているのはもう明らかなので、216行目のようなコメントは冗長でありない方が良いと思います。